doc/tutorial/argument_parser/index.md

```diff
@@ -0,0 +1,411 @@
+
+\htmlonly <details> <summary>Copy and paste this code into the beginning of your application: </summary>  \endhtmlonly
+\snippet doc/tutorial/argument_parser/solution3.cpp program
+\htmlonly </details> \endhtmlonly
+
+Your task is now to extend the initialization function by the following:
+
+1. Extend your `initialize_argument_parser` function by a parameter that takes a `cmd_arguments` object and adapt the function call in your main function to pass on `args`;
+2. Set the default value of `aggregate_by` to `"mean"`.
+
+You can now use the variables from `args` to add the following inside of the `initialize_argument_parser` function:
+
+2. Add a positional option to the parser that fills the variable `file_path` so our program knows the location of the data file to read in.
+3. Add an option `-y/--year` that fills the variable `year` which will enable our program to filter the data by only including a season if it got released after the value `year`.
+4. Add an option `-a/--aggregate-by` that fills the variable `aggregate_by` which will enable our program choose between aggregating by mean or median.
+5. Add a flag `-H/--header-is-set` that fills the variable `notation`  which let the program know if it should ignore the first line in the file.
+
+Take a look at the help page again after you've done all of the above. You will notice that your options have been automatically included. **Copy and paste the example data file from the introduction** and check if your options are set correctly by trying the following few calls:
+
+```console
+./game_of_parsing -H -y 2014 data.tsv
+7.9175
+```
+
+```console
+./game_of_parsing -H -y 2010 --aggregate-by median data.tsv
+6.84
+```
+\endassignment
+\solution
+\include doc/tutorial/argument_parser/solution3.cpp
+\endsolution
+
+# List options
+
+In some use cases you may want to allow the user to specify an option multiple times and store the values in a list. With the seqan3::argument_parser this behaviour can be achieved simply by choosing your input variable to be of a container type (e.g. std::vector). The parser registers the container type through the seqan3::container_concept and will adapt the parsing of command line arguments accordingly.
+
+Example:
+
+\snippet doc/tutorial/argument_parser/small_snippets.cpp option_list
+
+Adding this option to a parser will allow you to call the program like this:
+
+```console
+./some_program -n Jon -n Arya -n Ned
+```
+
+The vector `list_variable` will then contain all three names `["Jon", "Arya", "Ned"]`.
+
+## Positional list options?
+
+An arbitrary positional option cannot be a list because of the ambiguity of which value belongs to which positional option. We do allow the very last option to be a list though for convenience. Note that if you try to add a positional list option which is not the last positional option, a seqan3::parser_design_error will be thrown.
```

Comment:
```suggestion
An arbitrary positional option cannot be a list because of the ambiguity of which value belongs to which positional option. We do allow the very last option to be a list for convenience though. Note that if you try to add a positional list option which is not the last positional option, a seqan3::parser_design_error will be thrown.
```